packages/studio-ui/src/components/AddPageButton/StreamScopeCollector.tsx

```diff
@@ -12,26 +12,28 @@ export default function StreamScopeCollector({
   handleClose,
   handleConfirm,
 }: FlowStepModalProps) {
-  const [streamScope, setStreamScope] = useStreamScope();
+  const { state, actions } = useContext(AddPageContext);
+  const { streamScope } = state;
+  const { updateState } = actions;
 
   const onConfirm = useCallback(async () => {
-    if (streamScope === undefined) {
-      setStreamScope({});
-    }
+    updateState(streamScope ? { streamScope } : { streamScope: {} });
```

Comment:
Is this `updateState` necessary? From a first glance, it seems like this method would be invoking `updateState` with the `streamScope` that's already been set in state. If the `streamScope` is undefined, `updateSelection` should have already put `{ streamScope: {} }` in the overall state.

Reply:
I changed it to just set  `streamScope` to `{}` if a `streamScope` does not exist.  This is needed because if the user does not ever interact with the dropdown, the `updateSelection` will never be called, and `streamScope` will be `undefined` instead of `{}`!